pkg/deploy/deploy_rp.go

```diff
@@ -90,22 +95,70 @@ func (d *deployer) DeployRP(ctx context.Context) error {
 			continue
 		}
 		if err != nil {
+			if retry := d.rpRemoveFailedScaleset(ctx); retry {
+				continue
+			}
 			return err
 		}
-
 		break
 	}
 
 	return d.configureDNS(ctx)
 }
 
+func (d *deployer) convertToIPAddressOrRange(ipSlice []string) []mgmtdocumentdb.IPAddressOrRange {
+	var ips []mgmtdocumentdb.IPAddressOrRange
+	for _, v := range ipSlice {
+		ips = append(ips, mgmtdocumentdb.IPAddressOrRange{IPAddressOrRange: to.StringPtr(v)})
+	}
+	return ips
+}
+
+func (d *deployer) rpRemoveFailedScaleset(ctx context.Context) (retry bool) {
+	vmssName := "rp-vmss-" + d.version
```

Comment:
There's a `const rpVMSSPrefix` defined in `upgrade_rp.go` within the same package which should be used instead of redefining `"rp-vmss-"`

Reply:
Fixed